composer/utils/checkpoint.py

```diff
@@ -304,7 +306,9 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
 
             # Send each file to the appropriate rank
             for file_name in file_list:
-                if dist.get_local_rank() == 0:  # Only 1 rank per node needs to transfer file
+                if dist.get_local_rank() == 0 or (
+                    torch_dist.get_rank(shard_process_group) == 0
```

Comment:
@bigning can we instead update our `get_global_rank` function to accept a process group and forward to `get_rank`?

Reply:
updated!